Feature: Generated typescript definition files

anbinm-msft opened the pull request:
- Switched to ts-loader since Awesome TS Loader doesn't generate .d.ts files for interfaces
- Removed all export default as this caused invalid d.ts files when stitched together
- Generate and stitch all .d.ts files a single .d.ts file

msftclas commented:
Hi __@mttwc__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.microsoft.com.
    </span>

TTYL, MSBOT;

anbiniyar committed:
Rename folder to use proper casing

anbiniyar committed:
Export items properly

This is required for “ts-loader” to work

anbiniyar committed:
Don’t export default

Exporting default makes it difficult to stitch all the typescript declaration files together

anbiniyar committed:
Export TS declaration files

- Swap out awesome-typescript-loader with ts-loader due to a bug in AT that prevents generation
   of TS declaration files for interfaces
- Stitch together the generated TS declaration files
- Remove the temporary folder once generated

msftclas commented:
@anbinm-msft,
Thanks for your contribution as a Microsoft full-time employee or intern. You do not need to sign a CLA.
_Thanks,
Microsoft Pull Request Bot_

anbinm-msft commented:
Please feel free to just squash and merge :)
